components/needs_team/needs_team.tsx

```diff
@@ -1,64 +1,81 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Route, Switch} from 'react-router-dom';
 import iNoBounce from 'inobounce';
 
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+import {Team, TeamMembership} from 'mattermost-redux/types/teams';
+
 import {startPeriodicStatusUpdates, stopPeriodicStatusUpdates} from 'actions/status_actions.jsx';
 import {startPeriodicSync, stopPeriodicSync, reconnect} from 'actions/websocket_actions.jsx';
 import * as GlobalActions from 'actions/global_actions.jsx';
+
 import Constants from 'utils/constants';
 import * as UserAgent from 'utils/user_agent';
 import * as Utils from 'utils/utils.jsx';
+
 import {makeAsyncComponent} from 'components/async_load';
 const LazyBackstageController = React.lazy(() => import('components/backstage'));
 import ChannelController from 'components/channel_layout/channel_controller';
 
 const BackstageController = makeAsyncComponent(LazyBackstageController);
 
-let wakeUpInterval;
+let wakeUpInterval: number;
 let lastTime = Date.now();
 const WAKEUP_CHECK_INTERVAL = 30000; // 30 seconds
 const WAKEUP_THRESHOLD = 60000; // 60 seconds
 const UNREAD_CHECK_TIME_MILLISECONDS = 10000;
 
-export default class NeedsTeam extends React.Component {
-    static propTypes = {
-        params: PropTypes.object,
-        currentUser: PropTypes.object,
-        currentChannelId: PropTypes.string,
-        currentTeamId: PropTypes.string,
-        actions: PropTypes.shape({
-            fetchMyChannelsAndMembers: PropTypes.func.isRequired,
-            getMyTeamUnreads: PropTypes.func.isRequired,
-            viewChannel: PropTypes.func.isRequired,
-            markChannelAsReadOnFocus: PropTypes.func.isRequired,
-            getTeamByName: PropTypes.func.isRequired,
-            addUserToTeam: PropTypes.func.isRequired,
-            selectTeam: PropTypes.func.isRequired,
-            setPreviousTeamId: PropTypes.func.isRequired,
-            loadStatusesForChannelAndSidebar: PropTypes.func.isRequired,
-            loadProfilesForDirect: PropTypes.func.isRequired,
-        }).isRequired,
-        theme: PropTypes.object.isRequired,
-        mfaRequired: PropTypes.bool.isRequired,
-
-        /*
-         * Object from react-router
-         */
-        match: PropTypes.shape({
-            params: PropTypes.shape({
-                team: PropTypes.string.isRequired,
-            }).isRequired,
-        }).isRequired,
-        history: PropTypes.object.isRequired,
-        teamsList: PropTypes.arrayOf(PropTypes.object),
+declare global {
+    interface Window {
+        isActive: boolean;
+    }
+}
+
+type Props = {
+    currentUser: {
```

Comment:
`currentUser` doesn't appear to be required in the original spec.

Reply:
https://github.com/mattermost/mattermost-webapp/pull/4839/commits/6e4dd5c0e85debc5c48328b9b5a618bd6bc15a32